src/typeahead/dataset.js

```diff
@@ -269,9 +269,10 @@ var Dataset = (function() {
         // do not render the suggestions as they've become outdated
         if (!canceled && rendered < that.limit) {
           that.cancel = $.noop;
-          rendered += suggestions.length;
+          
           that._append(query, suggestions.slice(0, that.limit - rendered));
-
+          rendered += suggestions.length;
```

Comment:
Is this even necessary ? I don't think that `rendered` is used after this, so I think we can remove this line.